app/lib/katello/validators/content_view_puppet_module_validator.rb

```diff
@@ -3,21 +3,25 @@ module Validators
     class ContentViewPuppetModuleValidator < ActiveModel::Validator
       def validate(record)
         if record.uuid.blank? && (record.name.blank? || record.author.blank?)
-          invalid_parameters = _("Invalid puppet module parameters specified.  Either 'uuid' or 'name' and 'author' must be specified.")
-          record.errors[:base] << invalid_parameters
-          return
-        end
-
-        if record.name && record.author
-          # validate that a puppet module exists with this name+author
-          unless PuppetModule
-                  .in_repositories(record.content_view.puppet_repos)
-                  .where(:name => record.name, :author => record.author).present?
-
-            invalid_parameters = _("Puppet Module with name='%{name}' and author='%{author}' does not exist") %
-                { :name => record.name, :author => record.author }
-
-            record.errors[:base] << invalid_parameters
+          record.errors[:base] << _("Invalid puppet module parameters specified. \
+                                    Either 'uuid' or 'name' and 'author' must be specified.")
+        elsif record.name && record.author &&
+          !PuppetModule.exists?(name: record.name, author: record.author)
+          record.errors[:base] << _("Puppet Module with name='%{name}' and author='%{author}' does\
+                                    not exist") % { name: record.name, author: record.author }
+        elsif record.uuid && !PuppetModule.exists?(uuid: record.uuid)
+          record.errors[:base] << _("Puppet Module with uuid='%{uuid}' does not\
+                                    exist") % { uuid: record.uuid }
+        else
+          puppet_module = if record.uuid.blank?
+                            PuppetModule.find_by(name: record.name, author: record.author)
```

Comment:
So we need to validate that there is at least one PuppetModule with that name/author combination, but there may be duplicate PuppetModules with that name/author, right? `find_by` returns the last match (like `where().last`), so this validation does ensure that at least one PM exists, but it sets the uuid to the last PM's uuid. Please advise on how to handle this situation.